locations/spiders/mcdonalds_ph.py

```diff
@@ -0,0 +1,42 @@
+import re
+
+import chompjs
+import scrapy
+
+from locations.categories import Categories, Extras, apply_category, apply_yes_no
+from locations.dict_parser import DictParser
+from locations.hours import OpeningHours
+from locations.spiders.mcdonalds import McDonaldsSpider
+
+
+class McDonaldsPHSpider(scrapy.Spider):
+    name = "mcdonalds_ph"
+    item_attributes = McDonaldsSpider.item_attributes
+    start_urls = ["https://www.mcdonalds.com.ph/store-locator"]
+
+    def parse(self, response, **kwargs):
+        raw_data = response.xpath("//store-locator").get()
+        locations = chompjs.parse_js_object(
+            "".join(re.findall(r'<store-locator :stores="(.*?)></store-locator>', raw_data))[:-1].replace("&quot;", '"')
```

Comment:
Is join and find all required here?

Reply:
I think I used a join here because the input for chompj needs to be a string.  Used findall to easily reduce string between the two patterns.  There was some complex string manipulation required here.  Perhaps I could have used a re.search or different method?  It is returning the results I am looking for.  Did you have a different suggestion or see any errors?

Reply:
I think the first issue you had was the xpath (`:stores`), xpath, isn't great with namespaces, or any `:`. In this case we can avoid the drama with `attrib` on the selector.